src/Stratis.Bitcoin.Features.Consensus/ConsensusFeature.cs

```diff
@@ -209,26 +209,55 @@ public static IFullNodeBuilder UseStratisConsensus(this IFullNodeBuilder fullNod
                         services.AddSingleton<IConsensusLoop, ConsensusLoop>();
                         services.AddSingleton<StakeChainStore>().AddSingleton<StakeChain, StakeChainStore>(provider => provider.GetService<StakeChainStore>());
                         services.AddSingleton<IStakeValidator, StakeValidator>();
-                        services.AddSingleton<ConsensusManager>().AddSingleton<INetworkDifficulty, ConsensusManager>()
-                            .AddSingleton<IGetUnspentTransaction, ConsensusManager>();
+                        services.AddSingleton<ConsensusManager>().AddSingleton<INetworkDifficulty, ConsensusManager>().AddSingleton<IGetUnspentTransaction, ConsensusManager>();
                         services.AddSingleton<IInitialBlockDownloadState, InitialBlockDownloadState>();
                         services.AddSingleton<ConsensusController>();
                         services.AddSingleton<ConsensusStats>();
                         services.AddSingleton<ConsensusSettings>();
-                        services.AddSingleton<IConsensusRules, ConsensusRules>();
-                        services.AddSingleton<IRuleRegistration, CoreConsensusRules>();
+                        services.AddSingleton<IConsensusRules, PosConsensusRules>();
+                        services.AddSingleton<IRuleRegistration, PosConsensusRulesRegistration>();
                     });
             });
 
             return fullNodeBuilder;
         }
 
-        public class CoreConsensusRules : IRuleRegistration
+        public class PowConsensusRulesRegistration : IRuleRegistration
```

Comment:
Maybe consider adding a common class that should contain rules shared between PoW and PoS and the `PowConsensusRulesRegistration` and `PosConsensusRulesRegistration` would derive from it and extend the rules list?

Reply:
I did consider that indeed, I feel however that the set of rules are what uniquely define a blockchain. so in this case its important to know exactly what are the rules that are in play and the order they get registered, I would for sure consider putting rules in to functionality groups where it makes sense (like all rules that manage deployments, or context transaction checks) however this is really hard to get right in a sense that what are the boundaries of logical grouping of rules.